qiskit/transpiler/passes/scheduling/scheduling/alap.py

```diff
@@ -37,57 +37,16 @@ def run(self, dag):
             TranspilerError: if the circuit is not mapped on physical qubits.
             TranspilerError: if conditional bit is added to non-supported instruction.
         """
-        if len(dag.qregs) != 1 or dag.qregs.get("q", None) is None:
-            raise TranspilerError("ALAP schedule runs on physical circuits only")
+
         if self.property_set["time_unit"] == "stretch":
             raise TranspilerError("Scheduling cannot run on circuits with stretch durations.")
 
+        node_durations = {
+            node._node_id: self._get_node_duration(node, dag) for node in dag.topological_op_nodes()
+        }
```

Comment:
Hmm... I'm not quite sure you should you should be accessing node._node_id here, even though it is a python expose method. Perhaps you could keep the accessing the node_id in Rust instead of here. Afterall DAGNode is still a PyO3 struct and it stores an optional instance of `NodeIndex`, the rust native indexing type for DAG.